### PR TITLE
Add Equo Chromium implementation in equo ide

### PR DESCRIPTION
**Pull request by agustinschilling**
The objective is to offer users the option to run equoIde using Equo Chromium browsers (v106 OSS).

A new project (chromium-solstice) was added, which overrides some SWT classes from eclipse.

To use the jar that overrides the SWT classes, it's necessary to unsign the eclipse.swt jar. To do this I opted to generate a copy and unsign that one, storing it in ~/.equo/unsigned-jar by default.
This ensures that the added classes are used when creating instances of Browser, BrowserFunction, OpenWindowsListener, WindowEvent, while the flow works normally in other cases.

To use it, the user would need to add it to the config file as follows:

MAVEN: `(pom.xml)`
```
  <configuration>
    <equoChromium>true</equoChromium>
  </configuration>
```

GRADLE: `(build.gradle)`
```
  equoIde {
    equoChromium()
  }
```

### CODE REVIEW

**Comment by nedtwigg** [comment]
Sorry for long delay, this is great work, I'll push up some tweaks and comments later today.

**Comment by nedtwigg** [comment]
This works great! Summary of changes I made:

- switched from `equoChromium()` to `useChromium()` since we are already inside the `equoIde` block
- instead of publishing the `org.eclipse.swt.browser.Browser` override classes as a separate jar, they are just bundled as a jar into `solstice`, and extracted when necessary
- centralized the classpath magic for chromium override logic into `BuildPluginIdeMain`

IMO there are two things that need to get fixed before merging this

1) if you are running on macOS arm (lots of early adopters are!) then you get `Could not resolve com.equo:...macosx.aarch64:106.0.0`. We should either add `aarch64` to the open source repo, or add some warnings on mac arm.
2) the watermark has not been tailored to IDE users

image

Is there a way for users to remove the watermark? Do they do `useChromium('license-key-12989bu')` or something like that? Or is there GPL version that comes without watermark?

File: solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java [hunk 1]
```diff
@@ -120,6 +135,8 @@ public void launch() throws IOException, InterruptedException {
 			}
 			vmArgs.add("-Dorg.slf4j.simpleLogger.defaultLogLevel=" + (isBlocking ? "info" : "error"));
 
+			vmArgs.add("-Dchromium.args=--disable-web-security");
```
Comment (nedtwigg) [inline solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java:138]: I'm guessing this is needed for the Eclipse help system? We should only enable this if in equo mode, and if there's a way we can limit it somehow to be more secure that would be very good as well.

**Review by nedtwigg — CHANGES_REQUESTED** [review]

File: solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java (hunk 1 above)
Comment (nedtwigg) [reply solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java]: Can we remove this flag? Is Eclipse Help the driver for this flag?

Comment (agustinschilling) [reply solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java]: This flag is required for Chromium to be able to access the filesystem, also, in some cases for BrowserFunctions to work.

[https://docs.equoplatform.com/chromium/106.x/troubleshooting/common-issues.html# _browserfunctions_are_not_executed_in_local_files](url)

Comment (nedtwigg) [reply solstice/src/main/java/dev/equo/ide/BuildPluginIdeMain.java]: In https://github.com/equodev/equo-ide/pull/123/commits/0dfd3498f34a31a7010289742eb4bc022d55c918 I removed the flag by default and added docs to the README explaining how and when to add the flag (vmArgs is not actually implemented yet). Does that look good to you? Or do you think it needs to be the default?

**Review by nedtwigg — APPROVED** [review]

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 1.3.0` and `plugin-maven 1.2.0`.

**Pull request merged**